docs/_clef/Rules.md

```diff
@@ -48,98 +87,171 @@ function ApproveListing(req){
 }
 ```
 
-Whenever the external API is called (and the ruleset is enabled), the `signer` calls the UI, which is an instance of a ruleset-engine. The ruleset-engine
-invokes the corresponding method. In doing so, there are three possible outcomes:
-
-1. JS returns "Approve"
-  * Auto-approve request
-2. JS returns "Reject"
-  * Auto-reject request
-3. Error occurs, or something else is returned
-  * Pass on to `next` ui: the regular UI channel.
-
-A more advanced example can be found below, "Example 1: ruleset for a rate-limited window", using `storage` to `Put` and `Get` `string`s by key.
-
-* At the time of writing, storage only exists as an ephemeral unencrypted implementation, to be used during testing.
-
-### Things to note
-
-The Otto vm has a few [caveats](https://github.com/robertkrimen/otto):
+When a request is made via the external API, the logic flow is as follows:
+
+* Request is made to the `signer` binary using external API
+   
+* `signer` calls the UI - in this case the ruleset engine
+
+* UI evaluates whether the call conforms to rules in an attested rulefile
+
+* Assuming the call returns "Approve", request is signed.
+
+
+There are five possible outcomes from the ruleset engine that are 
+handled in different ways:
+ 
+| Return value      |    Action                                 |
+| ------------------| ----------------------------------------- |
+| Approve           | Auto-approve request       			    |
+| Reject            | Auto-reject request        				|
+| Error             | Pass decision to UI for manual approval   |
+| Unexpected value  | Pass decision to UI for manual approval   |
+| Nothing           | Pass decision to UI for manual approval   |
```

Comment:
I'd say that there are three outcomes. 

1. The literal string `Approve`. 
2. The literal string `Reject`. 
3. Anything else

Reply:
```suggestion
 
| Return value      |    Action                                 |
| ------------------| ----------------------------------------- |
| "Approve"           | Auto-approve request       			    |
| "Reject"            | Auto-reject request        				|
| Anything else             | Pass decision to UI for manual approval   |
```